### PR TITLE
1.6.0

### PR DESCRIPTION
**Pull request by lucasstarsz**
# FastJ 1.6.0

## Highlights
- [SLF4J](https://www.slf4j.org/) integration (currently `2.0.0-alpha-5)` in #127
- Fixed dozens of concurrency "random crashes and errors" in #161
- Added custom input event system for keyboard and mouse usage
- Added `origin`, `unit`, and directional vectors for `Point/Pointf` in #155
- added substantial logging to the engine
- added `EngineConfig` for easier engine configuration
- revamped Display system, added simple dialog system in #124
- added Drawable `lookAt` functionality
- added simple sprites and textures support in #117 
- added resource management system in #117
- added support for textures in `.psdf` in #117 
- added `.obj` and `.mtl` support in #117 
- removed game crashing errors in the audio engine

## Breaking Changes
- Deprecated `Point/Pointf.Origin`
- Deprecated `FastJEngine#init(gameTitle, gameManager, fps, ups, windowResolution, canvasResolution, hardwareAcceleration, exceptionAction)`
- original display system is no longer supported

## Other Changes
- Added [Spotless](https://github.com/diffplug/spotless) integration for import order (and hopefully more things later)
- Added trace-level logging and date/time recording to `test` logging
- moved `src/example` to its own Gradle subproject in #127
- fixed audio example sound usage

**Full Changelog**: https://github.com/fastjengine/FastJ/compare/1.5.1...1.6.0

### CODE REVIEW

**Comment by Sammie156** [comment]
Ah! The 1.6.0 update 😌 . After 1500 years of prediction, hard work, praying, fighting, studying and sorcery, it's here. I'll get my review done as soon as I can.

File: build.gradle [hunk 1]
```diff
@@ -49,49 +49,11 @@ wrapper.gradleVersion = '7.1.1'
...
-            )
-        }
-    }
-}
-
+dependencies.testImplementation(dependencies.platform('org.junit:junit-bom:5.8.2'))
+dependencies.testImplementation('org.junit.jupiter:junit-jupiter:5.8.2')
+dependencies.testRuntimeOnly("org.junit.platform:junit-platform-launcher")
+dependencies.api('org.slf4j:slf4j-api:2.0.0-alpha5')
+dependencies.testImplementation('org.slf4j:slf4j-simple:2.0.0-alpha5')
```
Comment (Sammie156) [inline build.gradle:56]: This is pretty cool and really nice.

Comment (lucasstarsz) [reply build.gradle:56]: Yeah! It's definitely been a life saver having a proper logging library.

**Comment by lucasstarsz** [comment]
For the time being, I'm not going to worry about hitting our target coverage. Increases over time as we strive towards a proper stable release of the engine is key 👍🏾

**Comment by Sammie156** [comment]
> For the time being, I'm not going to worry about hitting our target coverage. Increases over time as we strive towards a proper stable release of the engine is key 👍🏾

I agree completely!!

**Comment by YeffyCodeGit** [comment]
1.6.0 is here after such a long wait!

**Comment by AurumByte** [comment]
Finally! 1.6.0! I'm very hyped to see this great engine grow!

**Pull request merged**